File: business-service/src/main/java/com/dayagent/service/PlanRequest.java
package com.dayagent.service;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Builder;
import lombok.Data;

import java.util.List;
import java.util.Map;

/**
 * 发给 Python /generate-plan 的请求体
 *
 * Java 用驼峰命名, Python 用下划线命名.
 * @JsonProperty 让 Jackson 序列化时转为 Python 期望的字段名.
 */
@Data
@Builder
@JsonInclude(JsonInclude.Include.NON_NULL)  // 不序列化 null 字段，让 Python 用默认值
public class PlanRequest {
    @JsonProperty("user_id")
    private String userId;

    @JsonProperty("yesterday_summary")
    private String yesterdaySummary;

    private List<String> goals;       // 刚好同名, 不用转
    private String location;          // 刚好同名, 不用转

    @JsonProperty("force_refresh")
    private Boolean forceRefresh;     // 跳过缓存，强制重新抓取

    private List<ParcelInfo> parcels; // 用户未签收的快递列表

    @JsonProperty("user_settings")
    private Map<String, Object> userSettings;

    @Data
    @Builder
    @JsonInclude(JsonInclude.Include.NON_NULL)
    public static class ParcelInfo {
        @JsonProperty("tracking_no")
        private String trackingNo;

        private String carrier;       // 快递公司
        private String remark;        // 备注
    }
}

File: agent_service/schemas/plan_request.py
import os
from typing import Any

from pydantic import BaseModel, Field

_DEFAULT_LOCATION = os.getenv("DEFAULT_LOCATION", "南昌")


class ParcelInfo(BaseModel):
    """Java 传入的快递单号信息"""
    tracking_no: str = Field(..., description="快递单号")
    carrier: str = Field(..., description="快递公司，如顺丰、京东")
    remark: str = Field(default="", description="备注，如'耳机'")


class PlanRequest(BaseModel):
    """Java 端调用 /generate-plan 时传入的数据"""
    user_id: str = Field(..., description="用户 ID")
    yesterday_summary: str = Field(default="", description="昨日总结")
    goals: list[str] = Field(default_factory=list, description="当前活跃目标列表")
    location: str = Field(default=_DEFAULT_LOCATION, description="城市，用于查天气")
    force_refresh: bool = Field(default=False, description="跳过缓存，强制重新抓取")
    parcels: list[ParcelInfo] = Field(default_factory=list, description="用户未签收的快递列表")
    user_settings: dict = Field(default_factory=dict, description="用户配置的 API Key 等")


class ParcelStatus(BaseModel):
    """快递状态查询结果，返回给 Java 端用于更新数据库"""
    tracking_no: str
    carrier: str
    remark: str = ""
    state: str
    is_delivered: bool = False
    pickup_code: str = ""              # ← 加这行
    is_waiting_pickup: bool = False
    latest_context: str = ""
    latest_time: str = ""
    details: list[dict[str, str]] = Field(default_factory=list, description="完整物流轨迹")


class PlanResponse(BaseModel):
    """返回给 Java 端的规划结果"""
    plan: str = Field(..., description="今日规划正文")
    priorities: list[str] = Field(default_factory=list, description="优先级列表")
    warnings: list[str] = Field(default_factory=list, description="预警提醒，如作业截止")
    parcels: list[ParcelStatus] = Field(default_factory=list, description="最新的快递状态")
